theano/gof/tests/test_vm.py

```diff
@@ -221,6 +222,25 @@ def test_partial_function_output_keys():
     assert f(5, output_subset=['a'])['a'] == f(5)['a']
 
 
+def test_partial_function_with_updates():
+
+    def check_updates(linker_name):
+        x = tensor.lscalar('input')
+        y = theano.shared(1, name='global')
+        f = theano.function([x], [x, x + 34], updates=[(y, x + 1)], mode=Mode(
+            optimizer=None, linker=linker_name))
+        g = theano.function([x], [x - 6], updates=[(y, y + 3)], mode=Mode(
+            optimizer=None, linker=linker_name))
+        f(3, output_subset=[])
```

Comment:
Shoudn't it be [] and not None?
On Jul 19, 2016 8:06 AM, "Frédéric Bastien" <notifications@github.com>
wrote:

> In theano/gof/tests/test_vm.py
> <https://github.com/Theano/Theano/pull/4669#discussion_r71357276>:
>
> > @@ -221,6 +222,25 @@ def test_partial_function_output_keys():
> >      assert f(5, output_subset=['a'])['a'] == f(5)['a']
> >
> >
> > +def test_partial_function_with_updates():
> > +
> > +    def check_updates(linker_name):
> > +        x = tensor.lscalar('input')
> > +        y = theano.shared(1, name='global')
> > +        f = theano.function([x], [x, x + 34], updates=[(y, x + 1)], mode=Mode(
> > +            optimizer=None, linker=linker_name))
> > +        g = theano.function([x], [x - 6], updates=[(y, y + 3)], mode=Mode(
> > +            optimizer=None, linker=linker_name))
> > +        f(3, output_subset=[])
>
> assert the output of f at the same time, to be sure it is None.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4669/files/44e5c99301af15ba59069ab7e3668cab83e9fbd2..73620123abef7bd8d5b8c93db355f49c8e1316e4#r71357276>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAePFBqSsFunQLQD9mLBsEkREvPSgo8jks5qXOfngaJpZM4I-ZJL>
> .
>